ntbs-service/Services/AuthorizationService.cs

```diff
@@ -37,78 +36,99 @@ public async Task<bool> IsUserAuthorizedToManageAlert(ClaimsPrincipal user, Aler
             return userTbServiceCodes.Contains(alert.TbServiceCode);
         }
 
-        public IEnumerable<NotificationBannerModel> SetFullAccessOnNotificationBanners(
+        public void SetFullAccessOnNotificationBanners(
             IEnumerable<NotificationBannerModel> notificationBanners,
             ClaimsPrincipal user)
         {
             notificationBanners.ToList().ForEach(async n =>
             {
                 if(n.NotificationStatus != NotificationStatus.Legacy)
                 {
-                    n.FullAccess = await CanEditBannerModelAsync(user, n);
+                    n.ShowPadlock = await CanEditBannerModelAsync(user, n);
                 }
             });
-            return notificationBanners;
         }
 
-        public async Task<bool> CanEditBannerModelAsync(ClaimsPrincipal user, NotificationBannerModel notificationBannerModel)
+        public async Task<bool> CanEditBannerModelAsync(ClaimsPrincipal user,
+            NotificationBannerModel notificationBannerModel)
         {
-            var tbServiceCode = notificationBannerModel.TbServiceCode;
-            var tbServicePhecCode = notificationBannerModel.TbServicePHECCode;
-            var locationPhecCode = notificationBannerModel.LocationPHECCode;
-            return await AuthorizeUserAccessForTbServiceAndPhecCodes(user, tbServiceCode, locationPhecCode, tbServicePhecCode);
+            if (_filter == null)
+            {
+                _filter = await GetUserPermissionsFilterAsync(user);
+            }
+
+            if (_filter.FilterByTBService)
+            {
+                return _filter.IncludedTBServiceCodes.Contains(notificationBannerModel.TbServiceCode);
+            }
+
+            if (_filter.FilterByPHEC)
+            {
+                var allowedCodes = _filter.IncludedPHECCodes;
+                return allowedCodes.Contains(notificationBannerModel.TbServicePHECCode) ||
+                       allowedCodes.Contains(notificationBannerModel.LocationPHECCode);
+            }
+            return true;

```

Comment:
Hmm, that looks a bit scary - it looks like the default is to show everything if neither of the filters applies!!
If this is trying to cater for National Team. The `PermissionsFilter` does actually have a `UserType` variable on it - I wonder if we should just use that and switch on it, abandoning the `FilterByX` getters?